src/lang/kindOf.js

```diff
@@ -1,20 +1,12 @@
 define(function () {
 
-    var _rKind = /^\[object (.*)\]$/,
-        _toString = Object.prototype.toString,
-        UNDEF;
+    var _toString = Object.prototype.toString;
```

Comment:
Closure to _toString increases the scope lookup for _toString by 1, I'd prefer just using Object.prototype.toString directly in the function call, to save a pointer, and scope lookup

Reply:
On the other hand it avoids someone modifying Object.prototype.toSting (very unlikely, but still) and to property accesses. At least the second might go up the prototype chain (again probably very unlikely).

I'll change it though, seems good to me.